group/views.py

```diff
@@ -20,140 +20,166 @@
-            {'name':'contest', 's_time':'20177777', 'e_time':'20111010'},
-        ] 
-        return render(
-            request, 'group/viewall.html', {
-                'data_list': all_running_contest_list, 
-                'title': 'running contest',
-                'list_type': 'runContest',
-            })
-
-def get_ended_contest(request):
-        all_ended_contest_list = [
-            {'name':'archive', 'time':'20111010'},
-            {'name':'archive', 'time':'20140505'},
-            {'name':'archive', 'time':'20122222'},
-            {'name':'archive', 'time':'20133333'},
-            {'name':'archive', 'time':'20144444'},
-            {'name':'archive', 'time':'20166666'},
-            {'name':'archive', 'time':'20111010'},
-            {'name':'archive', 'time':'20140505'},
-            {'name':'archive', 'time':'20122222'},
-            {'name':'archive', 'time':'20133333'},
-            {'name':'archive', 'time':'20144444'},
-            {'name':'archive', 'time':'20166666'},
-        ] 
-        return render(
-            request, 'group/viewall.html', {
-                'data_list': all_ended_contest_list, 
-                'title': 'ended contest',
-                'list_type': 'endContest',
-            })
-
-def get_all_announce(request):
-        all_announce_list = [
-            {'id':'1', 'name':'announce', 'op':'drowsy', 'time':'20111010', 'content':'111'},
-            {'id':'2', 'name':'announce', 'op':'drowsy', 'time':'20140505', 'content':'222'},
-            {'id':'3', 'name':'announce', 'op':'drowsy', 'time':'20122222', 'content':'333'},
-            {'id':'4', 'name':'announce', 'op':'drowsy', 'time':'20133333', 'content':'444'},
-            {'id':'5', 'name':'announce', 'op':'drowsy', 'time':'20144444', 'content':'555'},
-            {'id':'6', 'name':'announce', 'op':'drowsy', 'time':'20166666', 'content':'666'},
-            {'id':'7', 'name':'announce', 'op':'drowsy', 'time':'20122222', 'content':'777'},
-            {'id':'8', 'name':'announce', 'op':'drowsy', 'time':'20111010', 'content':'888'},
-            {'id':'9', 'name':'announce', 'op':'drowsy', 'time':'20140505', 'content':'999'},
-            {'id':'10', 'name':'announce', 'op':'drowsy', 'time':'20122222', 'content':'aaa'},
-            {'id':'11', 'name':'announce', 'op':'drowsy', 'time':'20133333', 'content':'bbb'},
-            {'id':'12', 'name':'announce', 'op':'drowsy', 'time':'20144444', 'content':'ccc'},
-            {'id':'13', 'name':'announce', 'op':'drowsy', 'time':'20166666', 'content':'ddd'},
-            {'id':'14', 'name':'announce', 'op':'drowsy', 'time':'20122222', 'content':'eee'},
-        ] 
-        return render(
-            request, 'group/viewall.html', {
-                'data_list': all_announce_list, 
-                'title': 'announce',
-                'list_type': 'announce',
-            })
+
+logger = log.get_logger()
+
+def get_running_contest(request, group_id):
+        
+    try:
```

Comment:
I suggest to write a function for checking whether group exist or not. (it can avoid duplicated code)